### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by juliotrigo**
* Initial implementation of the library:
  - Add the ability to subscribe to events
  - Add the ability to subscribe to keys
  - Add the ability to subscribe to databases
* Add support for Python ``3.5``, ``3.6``, ``3.7``
* Add support for Nameko ``2.11``, ``2.12``, ``3.0``
* Add support for (Python) Redis ``2.10``, ``3.0``, ``3.1``, ``3.2``
* Add support for Redis ``4.0``
* CI pipeline
* Documentation

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+Usage
+-----
+
+This Nameko_ extension adds the ability to subscribe to events, keys and
+databases.
+
+Some event examples:
+
+- ``expire`` events fire for ``EXPIRE`` commands
```
Comment (gianchub) [inline README.rst]: s/fire/fired

Also below

File: nameko_rediskn/rediskn.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+        Args:
+            uri_config_key (str): Redis URI config key.
+            events (str or list(str)): one or more events to subscribe to.
+            keys (str or list(str)): one or more keys to subscribe to.
+            dbs (str or list(str)): one or more DBs to subscribe to.
+        """
+        self.uri_config_key = uri_config_key
+
+        if events is None:
```
Comment (gianchub) [inline nameko_rediskn/rediskn.py]: We could have a small helper for this 4 lines logic which we repeat 3 times, or maybe use a ternary expression?

**Review by gianchub — APPROVED** [review]
Just a couple of makeup comments, feel free to ignore.

**Comment by juliotrigo** [comment]
@gianchub Thanks for the feedback.
I've implemented the suggestions. This is now ready for another review.

**Review by gianchub — APPROVED** [review]
LGTM!

File: nameko_rediskn/rediskn.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                'Provide either `events` or `keys` to get notifications'
+            )
+            log.error(error_message)
+            raise ConfigurationError(error_message)
+
+        self._redis_uri = self.container.config['REDIS_URIS'][
+            self.uri_config_key
+        ]
+        redis_config = self.container.config.get('REDIS', {})
+        self._notification_events = redis_config.get('NOTIFICATION_EVENTS')
```
Comment (heindsight) [inline nameko_rediskn/rediskn.py]: I tend to prefer lower case keys in all but top-level config items. It's a matter of personal preference though, so if you want to keep it upper case, that's fine.

**Review by heindsight — APPROVED** [review]

File: nameko_rediskn/rediskn.py (hunk 2 above)
Comment (juliotrigo) [reply nameko_rediskn/rediskn.py]: Nice suggestion! I went with the ternary expression.

File: nameko_rediskn/rediskn.py (hunk 3 above)
Comment (juliotrigo) [reply nameko_rediskn/rediskn.py]: I'll make it lowercase. It seems more readable to me and we also have a few other lower case attribute names in the pipeline :)

**Review by heindsight — APPROVED** [review]

**Pull request merged**